v1/nix/pkgs/fetchPip/fetchPip.py

```diff
@@ -113,35 +121,55 @@ def pip(venv_path, *args):
 
     venv_path = Path(".venv").absolute()
     create_venv(venv_path)
-    pip(venv_path, "install", "--upgrade", f"pip=={PIP_VERSION}")
+    pip(
+        venv_path,
+        "install",
+        "--upgrade",
+        f"pip=={PIP_VERSION}",
+    )
 
     cafile = generate_ca_bundle(HOME / ".ca-cert.pem")
     wait_for_proxy(proxy_port, cafile)
 
-    optional_flags = [
+    flags = [
         PIP_FLAGS,
         ONLY_BINARY_FLAGS,
-        REQUIREMENTS_LIST,
+        "--proxy",
+        f"https://localhost:{proxy_port}",
+        "--progress-bar",
+        "off",
+        "--cert",
+        cafile,
+        "--cache-dir",
+        cache_path,
     ]
-    if REQUIREMENTS_FILES:
-        optional_flags += ["-r " + " -r ".join(REQUIREMENTS_FILES.split())]
     if NO_BINARY:
         optional_flags += ["--no-binary " + " --no-binary ".join(NO_BINARY.split())]
+    if WRITE_METADATA:
+        metadata_flags = ["--report", "/build/report.json"]
+
+    for req in REQUIREMENTS_LIST.split(" "):
```

Comment:
Something I completely overlooked so far. Won't we run into issues with spaces in requirements which contain markers etc. which leads to wrong splitting. Should we pass the requirements list as a JSON string?

Reply:
... or alternatively just switch to [__structuredAttrs](https://nixos.mayflower.consulting/blog/2020/01/20/structured-attrs/) all the way.

Reply:
It should work as expected by accident for those, as they will be joined by ` ` in L158 without any separators. But the current implementation would break if a requirement file name would include a ` `, because then we end up with an extra `-r`. 

__structuredAttrs is surely a good idea, but as thats not really related to `--report` and i believe it isn't much more broken than on main, I'd prefer to do this in a follow-up pr?